src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/Worker.java

```diff
@@ -465,7 +468,7 @@ public Worker(
         this(applicationName, recordProcessorFactory, config, streamConfig, initialPositionInStream, parentShardPollIntervalMillis,
                 shardSyncIdleTimeMillis, cleanupLeasesUponShardCompletion, checkpoint, leaseCoordinator, execService,
                 metricsFactory, taskBackoffTimeMillis, failoverTimeMillis, skipShardSyncAtWorkerInitializationIfLeasesExist,
-                shardPrioritization, Optional.empty(), Optional.empty(), DEFAULT_WORKER_STATE_CHANGE_LISTENER);
+                shardPrioritization, Optional.empty(), Optional.empty(), DEFAULT_WORKER_STATE_CHANGE_LISTENER, new ShardSyncer(new KinesisLeaseCleanupValidator()));
```

Comment:
Seems like the `ShardSyncer` is being created multiple times. Can you make sure to create it only once. Each `Worker` should have only 1 `ShardSyncer`.

Reply:
The ShardSyncer has only been instantiated in the constructors, so each worker has only 1 shard syncer. I have created a private static final instance instead now.